src/nvim/option.c

```diff
@@ -1486,8 +1403,21 @@ static void do_set_option(int opt_flags, char **argp, bool *did_show, char *errb
     }
   }
 
-  do_set_option_value(opt_idx, opt_flags, argp, prefix, nextchar, op, flags, varp,
-                      errbuf, errbuflen, errmsg);
+  // Don't try to change hidden option.
+  if (varp == NULL) {
+    return;
+  }
+
+  OptVal newval = get_option_newval(opt_idx, opt_flags, prefix, argp, nextchar, op, flags, varp,
```

Comment:
(off topic for this PR) AFAIR it's very unusual in the codebase to hide a pointer in a type. I.e. I would expect this to be:

```suggestion
  OptVal *newval = get_option_newval(opt_idx, opt_flags, prefix, argp, nextchar, op, flags, varp,
```

Is there a technical reason we didn't do this for `OptVal`?

Reply:
Well, it works similarly to the `Object` struct